third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/MemoryOpToLLVM.cpp

```diff
@@ -46,8 +48,7 @@ struct LocalLoadOpConversion
       auto needTrans = kOrder != sharedEnc.getOrder()[0];
       // Limitation 1 [TODO: remove]: Check LL bases to verify register and
       // address alignment
-      auto canUseLdmatrix =
-          (kWidth == vecWidth) && (!sharedEnc.getHasLeadingOffset());
+      auto canUseLdmatrix = (kWidth == vecWidth);
```

Comment:
@ThomasRaoux Is `getHasLeadingOffset` replaced by `NVMMASharedEncodingAttr` now? We are not sure if this path works fine actually using ldmatrix. At least we do not have solid test cases

Reply:
Oh, I get it now. We only accept `SwizzledSharedEncodingAttr` for this code path, which is fine to me

Reply:
> @ThomasRaoux Is getHasLeadingOffset replaced by NVMMASharedEncodingAttr now?

yesh basically `getHasLeadingOffset` has been weird for a while as it would totally change the meaning of the rest of the encoding and has only few hardcoded supported layouts (it's more a tiled layout as things are stored per 2D blocks and only support the layout supported by WGMMA/MMAv5). I don't think we had ever implemented ldmatrix for this layout but once you have the linear layout path it should work out of the box